homeworks/01-bubble-sort/src/App.tsx

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import "./App.css";
+import { ArrayCell, generateWidthArray } from "./arrayGeneration";
+
+function sort(
+  data: ArrayCell[]
+): { sortedData: ArrayCell[]; isSolved: boolean; currentId: number | null } {
+  for (let i = 0; i < data.length; i++) {
+    for (let j = 0; j < data.length - 1 - i; j++) {
+      if (data[j].value > data[j + 1].value) {
+        const temp = data[j];
+        data[j] = data[j + 1];
+
+        data[j + 1] = temp;
+        return {
+          sortedData: data.slice(),
+          isSolved: false,
+          currentId: data[j].id,
+        };
+      }
+    }
+  }
+  return { sortedData: data, isSolved: true, currentId: null };
+}
```

Comment:
I would say it should be in a separate file (eg, `utils` folder under `src`). Keep the component logic as clear as possible.